src/main/java/io/confluent/copycat/hdfs/ParquetRecordWriterProvider.java

```diff
@@ -44,12 +42,12 @@
     Path path = new Path(fileName);
     path.getFileSystem(conf).create(path);
     final ParquetWriter<GenericRecord> writer =
-        new AvroParquetWriter<GenericRecord>(path, avroSchema, compressionCodecName, blockSize, pageSize);
+        new AvroParquetWriter<>(path, avroSchema, compressionCodecName, blockSize, pageSize);
 
     return new RecordWriter<Long, SinkRecord>(){
       @Override
       public void write(Long key, SinkRecord record) throws IOException{
-        Object value = AvroData.convertToAvro(record.getValue());
+        Object value = AvroData.fromCopycatData(record.keySchema(),record.value());
```

Comment:
Why does this use `record.keySchema()` with `record.value()`? I think this code is ignoring keys, so shouldn't it be `record.valueSchema()` instead of `record.keySchema()`? I think the correct value is used earlier when you convert the the schema to Avro.